src/main/java/org/openstreetmap/atlas/checks/validation/points/DuplicatePointCheck.java

```diff
@@ -0,0 +1,72 @@
+package org.openstreetmap.atlas.checks.validation.points;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import org.openstreetmap.atlas.checks.base.BaseCheck;
+import org.openstreetmap.atlas.checks.flag.CheckFlag;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasObject;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.utilities.collections.Iterables;
+import org.openstreetmap.atlas.utilities.configuration.Configuration;
+import org.openstreetmap.atlas.utilities.scalars.Distance;
+
+/**
+ * This check looks for two or more {@link Point}s that are in the exact same location.
+ *
+ * @author savannahostrowski
+ */
+public class DuplicatePointCheck extends BaseCheck<Location>
+{
+    private static final List<String> FALLBACK_INSTRUCTIONS = Arrays
+            .asList("Nodes {0} are duplicates at {1}");
+    private static final Distance ZERO_DISTANCE = Distance.ZERO;
+    private static final long serialVersionUID = 8624313405718452123L;
+
+    @Override
+    protected List<String> getFallbackInstructions()
+    {
+        return FALLBACK_INSTRUCTIONS;
+    }
+
+    /**
+     * Default constructor
+     *
+     * @param configuration
+     *            the JSON configuration for this check
+     */
+    public DuplicatePointCheck(final Configuration configuration)
+    {
+        super(configuration);
+    }
+
+    @Override
+    public boolean validCheckForObject(final AtlasObject object)
+    {
+        return object instanceof Point;
+    }
+
+    @Override
+    protected Optional<CheckFlag> flag(final AtlasObject object)
+    {
+        final Point point = (Point) object;
+        this.markAsFlagged(point.getLocation());
```

Comment:
Would it make sense to flag only if we find duplicates? Otherwise, we're flagging at each location even if there is no duplicate there. If there's no duplicate there, we will never find another Point at that location. This would minimize memory impact. Maybe worth playing with.

Reply:
Actually, I'd agree. Will do.